components/backup-stream/src/checkpoint_manager.rs

```diff
@@ -208,6 +218,38 @@ impl CheckpointManager {
         self.notify(items.into_iter());
     }
 
+    /// "freeze" the current resolved ts to the checkpoint ts.
+    /// This is usually called before we are going to flush and after freezing
+    /// the current batch of mutations.
+    ///
+    /// When a flush of the data collector triggered:
+    ///
+    /// ```text
+    /// ----------------------|----------------->
+    ///                      ^^^
+    ///        Flushing data-+|+- Incoming data.
+    ///                       |
+    ///              Flush Freeze Tempfiles
+    /// ```
+    ///
+    /// Resolving over incoming data shouldn't advance the checkpoint of the
+    /// flushing data. So the current progress should be "freezed" when we are
+    /// about to flush.
+    pub fn freeze(&mut self) {
+        info!("log backup checkpoint manager freezing.";
+            "resolved_ts_len" => %self.resolved_ts.len(),
+            "resolved_ts" => ?self.get_resolved_ts(),
+            "frozen" => self.checkpoint_ts.len(),
+        );
+        self.checkpoint_ts = std::mem::take(&mut self.resolved_ts);
```

Comment:
I see there are many functions, such as `add_subscriber`, `get_from_region` and `get_all`, getting the checkpoint ts from `CheckpointManager.checkpoint_ts`.
After this PR, the caller has a high probability of getting checkpoint ts whose data hasn't been flushed.